plugin-influxdb/src/lib.rs

```diff
@@ -174,22 +161,60 @@ impl Output for InfluxDbOutput {
     }
 }
 
+// Returns true if the attribute with this key should be serialized as an InfluxDB tag,
+// false if it should become a field.
+fn partition_tag(
+    key: &str,
+    attributes_as: &AttributeAs,
+    attributes_as_tags: &HashSet<String>,
+    attributes_as_fields: &HashSet<String>,
+) -> bool {
+    match attributes_as {
+        AttributeAs::Tag => {
+            // default is tag => tag unless if in set
+            !attributes_as_fields.contains(key)
+        }
+        AttributeAs::Field => {
+            // default is field => tag only if in set
+            attributes_as_tags.contains(key)
+        }
+    }
+}
+
+// Check if the tag key is reserved and in case it is append a prefix
+fn ensure_reserved_tag_key(reserved_tags: [&str; 4], tag_key: String) -> String {
```

Comment:
Since the list of reserved tags is constant and depends on InfluxDB's spec, I think it would make more sense to use that constant in the function instead of passing it as a parameter.

Reply:
How about calling the function `ensure_valid_tag_key` instead? It tells more what it does: ensure that it's valid (not "ensure that it's reserved). (please don't hurt me for starting a naming war :smile:)